### PR TITLE
add unittest cover

### PR DESCRIPTION
**Pull request by yxxhero**
add unittest cover

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,7 +39,8 @@ build: lint
 
 .PHONY: test
 test:
-	go test -v ./...
+	go test -v ./... -coverprofile cover.out -race -p=1
```
Comment (mumoshu) [inline Makefile:42]: @yxxhero Why do we need -p=1?

Comment (yxxhero) [reply Makefile:42]: I don't know this flag. it comes from the internet. if you think that we don't need this. I will remove that. @mumoshu

Comment (mumoshu) [reply Makefile:42]: Go ahead!
On a side note, we may find `-covermode=atomic` useful once we start using `t.Parallel()`.
However we don't use Parallel today so we could just remove `-p=1` for now.

Comment (yxxhero) [reply Makefile:42]: got it.

**Pull request merged**